src/catalog/tests/open.rs

```diff
@@ -923,11 +923,20 @@ async fn test_concurrent_open(state_builder: TestCatalogStateBuilder) {
 
     // Open should be successful without retrying, even though there is an active state.
     let _state = state_builder
+        .clone()
         .unwrap_build()
         .await
         .open(SYSTEM_TIME(), &test_bootstrap_args())
         .await
         .unwrap();
 
     state_handle.await.unwrap();
+
+    // Open again to ensure that we didn't commit an invalid retraction.
+    let _state = state_builder
+        .unwrap_build()
+        .await
+        .open(SYSTEM_TIME(), &test_bootstrap_args())
+        .await
+        .unwrap();
```

Comment:
This test was already reproducing the issue, we just needed to re-read the catalog contents to see that it was corrupt.